wallet2/db/db.go

```diff
@@ -0,0 +1,549 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	//nolint:all
+	_ "github.com/glebarez/go-sqlite"
+)
+
+const (
+	AddressTable     = "addresses"
+	TransactionTable = "transactions"
+	PairTable        = "pairs"
+)
+
+type DB interface {
+	CreateTables() error
+
+	InsertIntoAddress(addr *Address) (*Address, error)
+	InsertIntoTransaction(t *Transaction) (*Transaction, error)
+	InsertIntoPair(key string, value string) (*Pair, error)
+
+	UpdateAddressLabel(addr *Address) (*Address, error)
+
+	GetAddressByID(id int) (*Address, error)
```

Comment:
I don't see any benefit by exporting ID or Index. Address string should be unique always.